rclcpp/include/rclcpp/publisher.hpp

```diff
@@ -169,7 +169,7 @@ class PublisherBase
 
   using IntraProcessManagerWeakPtr =
     std::weak_ptr<rclcpp::intra_process_manager::IntraProcessManager>;
-  bool use_intra_process_;
+  bool intra_process_is_enabled_;
```

Comment:
Regarding the naming of the variable I see why you don't want to use `use_` anymore since the decision about if it is actually being used is delayed. I think the order of the parts of the variable name was better before. I would suggest `allow_intra_process_` instead.

Reply:
I don't agree about this name change suggestion, but the other one is fine.

```c++
if (!allow_intra_process_ || has_inter_process_subscriptions) {
```

Isn't as clear to read in my opinion, as it reads to me one of "not allow intra process" or "allow not intra process", neither of which make sense to me, you have to look at it a while to get to "intra process is not allowed" and even then allow implies to me that it may or may not be enabled, which isn't the case. At this point it is either enable or it is not.

Whereas:

```c++
if (!intra_process_is_enabled_ || has_inter_process_subscriptions) {
```

Is very clear immediately, as it reads one of "not intra process is enabled" or "intra process is not enabled", which are better for me than the alternative you suggested.

So 👎 from me to change this variable name.